sdk/nodejs/appservice/zMixins_http.ts

```diff
@@ -61,6 +61,20 @@ export interface HttpHostSettings extends mod.HostSettings {
     }
 }
 
+export interface HttpFunctionArgs extends mod.CallbackArgs<mod.Context<HttpResponse>, HttpRequest, HttpResponse> {
+    /**
+     * Defines the route template, controlling to which request URLs your function responds. The
+     * default value if none is provided is <functionname>.
+     */
+    route?: pulumi.Input<string>;
+
+    /**
+     * An array of the HTTP methods to which the function responds. If not specified, the function
+     * responds to all HTTP methods.
+     */
+    methods?: pulumi.Input<pulumi.Input<string>[]>;
+}
+
 export type HttpEventSubscriptionArgs = util.Overwrite<mod.CallbackFunctionAppArgs<mod.Context<HttpResponse>, HttpRequest, HttpResponse>, {

```

Comment:
can we move away from .Overwrite here?  In our public APIs it becomes very confusing for people to know what's going on.

Reply:
Do you mean `export interface HttpEventSubscriptionArgs extends mod.CallbackFunctionAppArgs<mod.Context<HttpResponse>, HttpRequest, HttpResponse>, HttpFunctionArgs` ?

How do we deal with mismatching types (`resourceGroupName`)?

Reply:
I basically mean: just write out the full interface that is expected (optionally extending from an existing interface if appropriate).  It may mean duplication.  But it is super simple to understand and document.  This is what we've done in awsx btw.

Reply:
Okay, it's all over mixins, let me take this as a separate PR

Reply:
#283

Reply:
Sounds good. #resolved